include/swift/AST/Builtins.def

```diff
@@ -983,6 +986,34 @@ BUILTIN_MISC_OPERATION_WITH_SILGEN(CreateAsyncTask,
 BUILTIN_MISC_OPERATION_WITH_SILGEN(CreateAsyncTaskInGroup,
     "createAsyncTaskInGroup", "", Special)
 
+/// createAsyncTaskWithExecutor(): (
+///     Int,                            // flags
+///     Builtin.Executor,               // executor
+///     @escaping () async throws -> T  // function
+/// ) -> Builtin.NativeObject
+///
+/// Create a new asynchronous task future, given flags, a parent task,
+/// task group and a function to execute.
+BUILTIN_MISC_OPERATION_WITH_SILGEN(CreateAsyncTaskWithExecutor,
+    "createAsyncTaskWithExecutor", "", Special)
+
+/// createAsyncTaskWithExecutor(): (
+///     Int,                            // flags
+///     Builtin.RawPointer,             // group
+///     Builtin.Executor,               // executor
+///     @escaping () async throws -> T  // function
+/// ) -> Builtin.NativeObject
+///
+/// Create a new asynchronous task future, given flags, a parent task,
+/// task group and a function to execute.
+BUILTIN_MISC_OPERATION_WITH_SILGEN(CreateAsyncTaskInGroupWithExecutor,
+    "createAsyncTaskInGroupWithExecutor", "", Special)
+
+/// Build a Builtin.Executor value from an "ordinary" task executor
+/// reference.
+BUILTIN_MISC_OPERATION_WITH_SILGEN(BuildOrdinaryTaskExecutorRef,
+    "buildOrdinaryTaskExecutorRef", "n", Special)
```

Comment:
Did you end up needing these builtins for some reason?

Reply:
Hmmm, I needed them while I was emitting "get the task executor" in SIL in nonisolated functions but that we moved away from. I think the current approach is the right one (wanted to confirm with this review request) so let me remove all the not used things in a commit:

Reply:
Github seems to have lost my comment -- yes, we need these because: 

```
struct UnownedTaskExecutor {

  @inlinable
  public init<E: TaskExecutor>(ordinary executor: __shared E) {
    #if $BuiltinBuildTaskExecutor
    self.executor = Builtin.buildOrdinaryTaskExecutorRef(executor)
    #else
    fatalError("Swift compiler is incompatible with this SDK version")
    #endif
  }
}
```

We arguably could "wing it" and use the `buildOrdinaryExecutorRef` though I'm unsure about the proliferation of pretending as if they're the same... Internally the representation is the same though, that is true -- the memory layout of SerialExecutorRef and TaskExecutorRef is the same, at least currently.